js/common/bridge.go

```diff
@@ -280,3 +280,72 @@ func Bind(rt *goja.Runtime, v interface{}, ctxPtr *context.Context) map[string]i
 
 	return exports
 }
+
+// TODO move this
+// ModuleInstance is what a module needs to return
+type ModuleInstance interface {
+	ModuleInstanceCore
+	GetExports() Exports
+}
+
+func getinterfaceMethods() []string {
+	var t ModuleInstance
+	T := reflect.TypeOf(&t).Elem()
+	result := make([]string, T.NumMethod())
+
+	for i := range result {
+		result[i] = T.Method(i).Name
+	}
+
+	return result
+}
+
+// ModuleInstanceCore is something that will be provided to modules and they need to embed it in ModuleInstance
+type ModuleInstanceCore interface {
+	// we can add other methods here
+	// sealing field will help probably with pointing users that they just need to embed this in the
+	GetContext() context.Context
+}
+
+type Exports struct {
+	Default interface{}
+	Others  map[string]interface{}
+}
+
+func GenerateExports(v interface{}) Exports {
+	exports := make(map[string]interface{})
+	val := reflect.ValueOf(v)
+	typ := val.Type()
+	badNames := getinterfaceMethods()
+outer:
+	for i := 0; i < typ.NumMethod(); i++ {
+		meth := typ.Method(i)
+		for _, badname := range badNames {
+			if meth.Name == badname {
+				continue outer
+			}
+		}
+		name := MethodName(typ, meth)
+
+		fn := val.Method(i)
+		exports[name] = fn.Interface()
+	}
+
+	// If v is a pointer, we need to indirect it to access fields.
+	if typ.Kind() == reflect.Ptr {
+		val = val.Elem()
+		typ = val.Type()
+	}
+	var mic ModuleInstanceCore // TODO move this out
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		if field.Type == reflect.TypeOf(&mic).Elem() {
+			continue
+		}
+		name := FieldName(typ, field)
+		if name != "" {
+			exports[name] = val.Field(i).Interface()
+		}
+	}
+	return Exports{Default: exports, Others: exports}
```

Comment:
Currently(using common.Bind), this is the behaviour so this is just backwards compatible.
Default and others is the difference between:
- `import metrics from "k6/metrics"` - here whatever is default will be `metrics`
- `import * as metrics from "k6/metrics"` - here all things in Others will be `metrics`
With common.BInd there is just no way to differentiate between them, now there is and if the user wants they can defined `Default` as being `Counter` for example (in the case of metrics there aren't good examples :rofl: ) so that 
```
import theDefaultOfMetrics, { Counter} from "k6/metrics"
var c = new theDefaultOfMetrics("coolcounter") // the same as 
var c = new Counter("coolcounter") // this
```
(I did actually test it and it works). 
For some other modules this will likely make more sense, and is actually a functionality of ESM that we should probably support, now, instead of when we start needing it ;).

You can probably use it in the `k6/execution` so you can only have a `default` import ? :thinking: 

But either way `GenerateExports` tries to be just the most close thing to `common.Bind`s behaviour currently.